src/views/Settings.vue

```diff
@@ -140,23 +140,29 @@ export default defineComponent({
     })
   },
   methods: {
-    setFacility (facility: any) {
-      if (this.userProfile) {
+    setFacility (event: any) {
+      // adding check for this.currentFacility.facilityId as it gets set to undefined on logout
+      // but setFacility is called again due to :value="currentFacility.facilityId" in ion-select
+      if (this.userProfile && this.currentFacility.facilityId && event.detail.value != this.currentFacility.facilityId ) {
+        const facilityId = this.currentFacility.facilityId;
         this.store.dispatch('user/setFacility', {
-          'facility': this.userProfile.facilities.find((fac: any) => fac.facilityId == facility['detail'].value)
+          'facility': this.userProfile.facilities.find((fac: any) => fac.facilityId == event.detail.value)
         });
+        if (Object.keys(this.uploadProducts).length > 0) {
```

Comment:
I think we should not do this way. We are changing the facility, asking the user to confirm, and reverting if the user deny. We may have limitation with select with current code, what I would suggest we have a data property for currentFacilityId, use it as model and update it with value change

Reply:
Updated sir